src/compositions/useFlowRuns.ts

```diff
@@ -33,7 +34,13 @@ export function useFlowRuns(flowRunIds: string[] | Ref<string[] | null | undefin
   })
 
   const subscription = useSubscriptionWithDependencies(api.flowRuns.getFlowRuns, parameters)
-  const flowRuns = computed(() => subscription.response)
+  const flowRunsResponse = computed(() => subscription.response)
```

Comment:
Just making sure I understand correctly.

So every component that calls for flow runs will still create a request, but these requests, if there are more than one at a time, get bundled together and duplicates get stripped at that point. Each request updates the storage, and if a new component causes a storage update then that would update all subscribers to the shared ref.

Reply:
This has been reworked a bit. But I'll still answer the question to make sure we're all on the same page. A couple of things are working together here. Subscriptions are still being used. So if multiple components ask for the same individual flow run a single request will be made. 

Here's what happens
1. subscription is created
2. response from subscription is added to storage
3. flow run from storage is returned to component

The response from those subscriptions will be added into storage. A watch is used to pick up the transition from `undefined` (api request hasn't completed yet) to `FlowRun` (api request completed). TECHNICALLY that same flow run can trigger `storage.add` to be called once per component. But they all have the same version of the flow run and there's not much of a cost of re-adding things to the storage. 

The storage class and the `useStorage` composition work together to automatically track which flow runs are being used by components and and removes flow runs from storage when they are no longer being used.